config/rewards.go

```diff
@@ -0,0 +1,138 @@
+}
+
+// RewardsUpdates returns an updated number of algos, total rewards, and rewards
+// base to reflect rewards up to some rewards level.
+func RewardsUpdates(
+	proto ConsensusParams, status basics.Status, microAlgosIn basics.MicroAlgos, rewardedMicroAlgosIn basics.MicroAlgos, rewardsBaseIn uint64, rewardsLevelIn uint64,
+) (basics.MicroAlgos, basics.MicroAlgos, uint64) {
+	if status == basics.NotParticipating {
+		return microAlgosIn, rewardedMicroAlgosIn, rewardsBaseIn
+	}
+
+	var ot basics.OverflowTracker
+	rewardsUnits := proto.RewardUnits(microAlgosIn)
+	rewardsDelta := ot.Sub(rewardsLevelIn, rewardsBaseIn)
+	rewards := basics.MicroAlgos{Raw: ot.Mul(rewardsUnits, rewardsDelta)}
+	microAlgosOut := ot.AddA(microAlgosIn, rewards)
+	if ot.Overflowed {
+		panic(fmt.Sprintf("RewardsUpdates(): overflowed account balance when applying rewards %v + %d*(%d-%d)", microAlgosIn, rewardsUnits, rewardsLevelIn, rewardsBaseIn))
+	}
+	rewardsBaseOut := rewardsLevelIn
+	// The total reward over the lifetime of the account could exceed a 64-bit
+	// value. As a result this rewardAlgos counter could potentially roll over.
+	rewardedMicroAlgosOut := basics.MicroAlgos{Raw: rewardedMicroAlgosIn.Raw + rewards.Raw}
+	return microAlgosOut, rewardedMicroAlgosOut, rewardsBaseOut
+}
+
+// WithUpdatedRewards returns an updated number of algos in an AccountData
+// to reflect rewards up to some rewards level.
+func (proto ConsensusParams) WithUpdatedRewards(u basics.AccountData, rewardsLevel uint64) basics.AccountData {
+	u.MicroAlgos, u.RewardedMicroAlgos, u.RewardsBase = RewardsUpdates(
+		proto, u.Status, u.MicroAlgos, u.RewardedMicroAlgos, u.RewardsBase, rewardsLevel,
+	)
+
+	return u
+}
+
+// NormalizedOnlineBalance returns a “normalized” balance for this account.
+//
+// The normalization compensates for rewards that have not yet been applied,
+// by computing a balance normalized to round 0.  To normalize, we estimate
+// the microalgo balance that an account should have had at round 0, in order
+// to end up at its current balance when rewards are included.
+//
+// The benefit of the normalization procedure is that an account's normalized
+// balance does not change over time (unlike the actual algo balance that includes
+// rewards).  This makes it possible to compare normalized balances between two
+// accounts, to sort them, and get results that are close to what we would get
+// if we computed the exact algo balance of the accounts at a given round number.
+//
+// The normalization can lead to some inconsistencies in comparisons between
+// account balances, because the growth rate of rewards for accounts depends
+// on how recently the account has been touched (our rewards do not implement
+// compounding).  However, online accounts have to periodically renew
+// participation keys, so the scale of the inconsistency is small.
+func (proto ConsensusParams) NormalizedOnlineBalance(u basics.AccountData) uint64 {
```

Comment:
Couldn't this function be on a type defined in `basics` that gets instantiated in config? For whatever reason, this function in particular feels out of place being inside of config..